### PR TITLE
ci: add workflow to update go versions

### PR DESCRIPTION
**Pull request by cwaldren-ld**

### CODE REVIEW

**Review by kinyoklion — APPROVED** [review]
Is there some magic to update the readme and go.mod? (just question/suggestion, not a PR change request.)

**Comment by cwaldren-ld** [comment]
No. I will add such magic.

File: .github/workflows/check-go-versions.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          sed -e "s#latest=[^ ]*#latest=${{ env.officialLatestVersion }}#g"
+              -e "s#penultimate=[^ ]*#penultimate=${{ env.officialPenultimateVersion }}#g"
+              ./.github/variables/go-versions.env
+          sed -i "s/Go version {{ steps.go-versions.outputs.penultimate }}/Go version ${{ env.officialPenultimateVersion }}/g" README.md
+
+      - name: Update go.mod and testservice/go.mod
+        if: steps.update-go-versions.outcome == 'success'
+        id: update-go-mod
+        run: |
+          go mod edit -go=${{ env.officialPenultimateVersion }}
```
Comment (kinyoklion) [inline .github/workflows/check-go-versions.yml:67]: Fancy.

**Comment by cwaldren-ld** [comment]
This'll fail on the v6 branch since the README isn't in exactly the same format. At least, it should fail. I'll put up a PR for that shortly.

**Pull request merged**